src/components/response_error.tsx

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2018 Ultimaker B.V.
+import * as React from 'react';
+
+// components
+import Popup from './popup';
+import Tile from './tile';
+
+// utils
+import { I18n } from '../utils/i18n';
+
+/**
+ * Map of all API error codes that should be translated.
+ * The variables "code", "field_name" and "title" are available here.
+ */
+export const ERROR_CODE_TO_I18NC_TEMPLATE = {
+
+    // generic errors
+    "genericError": "There was a problem on the server, please try again. (Code: %{code})",
+    "apiResponseError": "There was a problem on the server, please try again.",
+
+    // generic field errors
+    "fieldError": "%{field_name} was invalid: %{title}.",
+    "lessThanMinimum": "%{field_name} does not have enough characters.",
+
+    // specific field errors
+    "emailInvalid": "Email address was not valid. Please use only lower-case characters and a full domain name.",
+    "fullNameInvalid": "Full name was not valid. Please use only latin characters, \
+        numbers and simple separators (single space, underline or hyphen).",
+    "usernameInvalid": "Username was not valid. Please use only latin characters, \
+        numbers and simple separators (single space, underline or hyphen).",
+    "passwordInvalid": "Password was not valid. Please use at least eight characters, \
+        at least one number and both lower and uppercase letters and special characters.",
+    "badPasswordRepeat": "Repeated password did not match original password.",
+    "dataUriImageInvalid": "Profile image was not valid. Please use a JPEG or PNG image."
+
+}
+
+/**
+ * Map of all form field names that should be translated.
+ */
+export const FIELD_NAME_TO_I18NC = {
```

Comment:
this should not be in the `response_error` file. We should have a separate file with translations